packages/flutter/lib/src/cupertino/tableview.dart

```diff
@@ -0,0 +1,485 @@
+  /// [plainChildCount] is not provided, then all plain children will be
+  /// created from the very beginning.
+  final int plainChildCount;
+
+  @override
+  _CupertinoTableViewState createState() => _CupertinoTableViewState();
+}
+
+class _CupertinoTableViewState extends State<CupertinoTableView> {
+  // The size of the last cell in the table, used to determine the desired gap
+  // between all extra dividers below it.
+  Size _lastItemSize;
+
+  // Decorator that builds cell widgets with extra chrome and/or functionality,
+  // e.g., adding a divider to a cell.
+  IndexedWidgetBuilder _cellDecorator;
+
+  @override
+  void initState() {
+    super.initState();
+    _cellDecorator = _createCellDecorator();
+  }
+
+  @override
+  void didUpdateWidget(CupertinoTableView oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    _cellDecorator = _createCellDecorator();
+  }
+
+  @override
+  void reassemble() {
+    super.reassemble();
+    _cellDecorator = _createCellDecorator();
+  }
+
+  IndexedWidgetBuilder _createCellDecorator() {
+    if (widget.plainChildren != null) {
+      return (BuildContext context, int index) {
+        if (index >= widget.plainChildren.length) {
+          return null;
+        }
+
+        final bool isLastCell = index == widget.plainChildren.length - 1;
+        return _decorateCell(widget.plainChildren[index], isLastCell);
+      };
+    } else if (widget.plainChildrenBuilder != null) {
+      return (BuildContext context, int index) {
+        final Widget cell = widget.plainChildrenBuilder(context, index);
+        if (cell == null) {
+          return null;
+        }
+
+        bool isLastCell;
+        if (widget.plainChildCount != null) {
+          // The child count was provided so we will simply compare cell indices.
+          isLastCell = index == widget.plainChildCount - 1;
+        } else {
+          // No child count was provided, so the only way to know if this is the
+          // last cell is to build the next cell and check for null.
+          final Widget nextCell = widget.plainChildrenBuilder(context, index + 1);
+          isLastCell =  nextCell == null;
+        }
+
+        return _decorateCell(cell, isLastCell);
+      };
+    } else {
+      throw Exception('No children found.');
+    }
+  }
+
+  // Decorates an individual cell with a divider beneath it. If this cell
+  // [isLastCell] in the table then this decorator also adds a
+  // [_LayoutSizeReport] so that the height of this last cell can be reported
+  // to the extra dividers sliver.
+  Widget _decorateCell(Widget cell, bool isLastCell) {
+    final Widget cellWithDivider = _CupertinoTableViewDivider(
+      child: cell,
+    );
+
+    if (isLastCell) {
+      return _LayoutSizeReport(
+        sizeCallback: (Size size) {
+          if (size != _lastItemSize) {
+            WidgetsBinding.instance.addPostFrameCallback((Duration dt) {
+              setState(() {
+                _lastItemSize = size;
```

Comment:
We're one frame behind? Wouldn't this mean in your tests where you just pump once, there would never be any extra dividers? Can we draw the extra dividers in the paint phase of a SingleChildRenderObjectWidget from _CupertinoTableViewDivider(isLastCell:, child:,) (by which point you already know the size of the last cell) in the same frame instead of the next frame?

Reply:
Can you elaborate on what you're suggesting?  I didn't follow.  I'm not sure what the context is around "SingleChildRenderObjectWidget".

Reply:
PS - This is why tests related to extra dividers use a pumpAndSettle() instead of just a pump().  I believe I commented on such.

Reply:
I changed this to a _CellHeightController with ChangeNotifier which no longer requires a `setState()` call.